Library/Homebrew/cmd/update.sh

```diff
@@ -220,8 +220,15 @@ homebrew-update() {
   for option in "$@"
   do
     case "$option" in
-      # TODO: - `brew update --help` should display update subcommand help
-      --help) brew --help; exit $? ;;
+      --help)
+        sed -n '/^  \* `update`/,/^  \* / {
+          s/`//g;               # Strip out backticks
+          s/\* update/update/;  # Mark the headers we want to keep
+          /\*/d;                # and delete the others
+          s/^  //;              # Adjust the spacing a bit
+          p;
+        }' "$HOMEBREW_LIBRARY/Homebrew/manpages/brew.1.md"
+      exit 0 ;;
```

Comment:
This is not performance-critical and we don't have to implement this part in Bash, so I think it's best to keep this simple. Meaning: let's just make this redirect to `brew help update` instead of `brew --help` and let `brew help <command>` figure out where to get the help text from (so we have a single place that deals with fetching the help text).